nodebuilder/tests/share_test.go

```diff
@@ -0,0 +1,211 @@
+//go:build share || integration
+
+package tests
+
+import (
+	"context"
+	"math"
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	libshare "github.com/celestiaorg/go-square/v2/share"
+
+	"github.com/celestiaorg/celestia-node/blob"
+	"github.com/celestiaorg/celestia-node/nodebuilder/node"
+	"github.com/celestiaorg/celestia-node/nodebuilder/tests/swamp"
+	"github.com/celestiaorg/celestia-node/share/shwap"
+	"github.com/celestiaorg/celestia-node/state"
+)
+
+func TestShareModule(t *testing.T) {
```

Comment:
This test is structured a lot like a unit test, which is helpful but not exactly what integration tests(Swamp) are for. Like intergration tests should not test "invalid height" that's a unit test concern. On the other side, this test would benefit from testing GetRow on Q1/Q2 and Q3/Q4 and accessing it from every node type, rather than light only. In simple terms, you want to test things that involve multiple components here(the full node), not the single ifs in one component. 

So as actionable I would make sure  the tests are working for every node type and move out unit tests out. If moving out units is complicated, I am fine with parking it in the issue.

Reply:
Thanks for the structured feedback. Make a lot of sense. Removing unit tests is not a big deal, so I'm fine w removing them.